storage/oqgraph/ha_oqgraph.cc

```diff
@@ -1099,6 +1099,10 @@ int ha_oqgraph::info(uint flag)
 
 int ha_oqgraph::extra(enum ha_extra_function operation)
 {
+  if (graph->get_thd() != current_thd) {
```

Comment:
I'd suggest to use ha_thd() instead of current_thd here as it normally returns table->in_use, which is cheaper than current_thd.

The trace Elena provided is generated by debug build. In release build tc_acquire_table() won't call this particular extra(HA_EXTRA_IS_ATTACHED_CHILDREN). The question is what if we call another handler method before calling any extra() in release build? Though I didn't check if that's the case.